### PR TITLE
Adds syndie tape for sec, atmos, and engineering.

### PR DESCRIPTION
**Pull request by ShiftyRail**
Webm of it in action : https://cdn.discordapp.com/attachments/246096207523348481/443154303062900736/tape.webm
Cost 8, 3 uses.
Right now, it explodes on unauthorised access, but via domino effect crossing one line can make all the tape explodes in one go.
When close to it, it has a different examine message as a heads-up.

:cl:
- rscadd: The syndicate can now provide security, atmospheric technicians, and engineers with rigged tape. Security rigged tape cuffs people who try to remove it, engineering rigged tape shocks you, and atmospherics tape is very sharp and difficult to remove.

### CODE REVIEW

**Comment by jknpj** [comment]
HOW ABOUT A TAPE THAT CAN BE BROKEN BY SIMPLE MOBS, HUH?

**Comment by ShiftyRail** [comment]
@jknpj [[[they]]] won't let me code that

File: code/WorkInProgress/Ported/policetape.dm [hunk 1]
```diff
@@ -181,12 +181,16 @@
 /obj/item/tape/attack_paw(mob/user as mob)
 	breaktape(/obj/item/weapon/wirecutters,user)
 
-/obj/item/tape/proc/breaktape(obj/item/weapon/W as obj, mob/user as mob)
+/obj/item/tape/proc/breaktape(obj/item/weapon/W as obj, mob/user as mob)
```
Comment (DamianX) [inline code/WorkInProgress/Ported/policetape.dm]: what the hell man

**Comment by MadmanMartian** [comment]
>>>>>[[[they]]]

File: code/WorkInProgress/Ported/policetape.dm (hunk 1 above)
Comment (ShiftyRail) [reply code/WorkInProgress/Ported/policetape.dm]: I don't even know

**Comment by SonixApache** [comment]
Blow their hand off instead of causing hull breaches?

>Maybe giving the option to put it on a foe's mouth to silence him and make him explode afterwards ?

The former of that would be a fantastic feature to have in general. The latter... maybe for syndie shit, sure.
also port tg/baycode yellow intent

**Comment by ShiftyRail** [comment]
@SonixApache
Security syndie tape is the most expensive and causes explosion
Atmos tape is sparks and shock (useful to light plasma on fire ?)
Engineering tape is so sharp it cuts off hands, and can't be remove easily (need an object with sharpness) and a force superior to a certain threshold 

Might want to change engineering and police, as it feels a bit more thematic

**Comment by DamianX** [comment]
Doesn't this make the tape explode twice, sometimes? `detonate` creates an explosion on the tile, which calls `ex_act` on the tape, which calls `detonate`...

**Comment by ShiftyRail** [comment]
@DamianX didn't do so when I tested - only one explosion in admin logs
The idea was to have dominos effects, with two tapes in quick succession, but it's actually not strong enough to make the adjacent tapes tiles explode on the same taperoll.

It should still explode when you toss a IED at them.

File: code/datums/uplink_item.dm [hunk 2]
```diff
@@ -344,6 +351,13 @@ var/list/uplink_items = list()
 	cost = 12
 	job = list("Station Engineer", "Chief Engineer")
 
+/datum/uplink_item/jobspecific/syndietape_engineering
+	name = "Syndicate Tape"
+	desc = "A length of enginerring tape rigged with explosives that will cuts the hands of trespassers. Very difficult to remove. Can be used 3 times."
```
Comment (Sooose) [inline code/datums/uplink_item.dm]: 'enginerring' should be 'engineering'

File: code/datums/uplink_item.dm [hunk 3]
```diff
@@ -352,6 +366,14 @@ var/list/uplink_items = list()
 	cost = 6
 	job = list("Atmospheric Technician", "Chief Engineer")
 
+/datum/uplink_item/jobspecific/syndietape_atmos
+	name = "Syndicate Tape"
+	desc = "A length of atmospherics tape charged with a powerful electric potential. Will shock people and spark who attempt to remove it, creating fires. Can be used 3 times."
```
Comment (Sooose) [inline code/datums/uplink_item.dm]: Should be "Will spark and shock people who attempt to remove it, " to fix my grammar autism.

**Comment by gbasood** [comment]
Hull breach tape? Hmmm

**Comment by DamianX** [comment]
I like the idea but I hate the execution

- I'd prefer if the security tape wasn't another DUDE BOMBS LMAO
- The shocking effect sounds more like an engineering thing to me, they already get the power gloves following this theme
- This leaves the third effect which is okay-ish but probably should be moved to the atmospherics one?

**Comment by Sooose** [comment]
I support Damian soley because engineering gets all the tator items that are actually fun,

**Comment by gbasood** [comment]
What if the tape cuffed its victim instead of exploding?

**Comment by SonixApache** [comment]
Wraps itself around the hands of non-security staff that touches it?

**Comment by ShiftyRail** [comment]
I fixed the grammar, the conflicts, but I'm not decided on what the sec tape should do
Emote 👍 for explosions and ❤️ for cuffing, so I can avoid responsibility and making choices

**Comment by ShiftyRail** [comment]
Here goes by popular request
100% tested

**Comment by SonixApache** [comment]
`Adds syndie tape for sec, atmos, and egineering. #18227`
`egineering`

File: code/WorkInProgress/Ported/policetape.dm [hunk 4]
```diff
@@ -214,5 +219,148 @@
...
+	if (get_dist(user, src) < 3 && charged)
+		to_chat(user, "<span class = 'warning'>This one has strange reflects.</span>")
+
+/obj/item/tape/engineering/syndie/emp_act(severity)
+	charged = FALSE
+	spark(src, 5)
+
+// Atmos syndie tape : hard to break and cut off your hands
+
+/obj/item/tape/atmos/atmos/destroy_tape(var/mob/user, var/obj/item/weapon/W)
```
Comment (DamianX) [inline code/WorkInProgress/Ported/policetape.dm]: \>atmos/atmos

File: code/datums/uplink_item.dm [hunk 5]
```diff
@@ -344,6 +351,13 @@ var/list/uplink_items = list()
 	cost = 12
 	job = list("Station Engineer", "Chief Engineer")
 
+/datum/uplink_item/jobspecific/syndietape_engineering
+	name = "Syndicate Tape"
+	desc = "A length of engineering tape charged with a powerful electric potential. Will spark and shock people who attempt to remove it, creating fires. Can be used 3 times."
+	item = /obj/item/taperoll/syndie/engineering
+	cost = 4
+	job = list("Security Officer", "Warden", "Head of Security")
```
Comment (DamianX) [inline code/datums/uplink_item.dm]: 🤔

File: code/WorkInProgress/Ported/policetape.dm [hunk 6]
```diff
@@ -214,5 +219,148 @@
...
+		return shock(user, 50)
+	else // No user, or not charged
+		return FALSE
+
+/obj/item/tape/engineering/syndie/proc/shock(var/mob/user, var/damage)
+	if (!user || !istype(user, /mob/living))
+		return FALSE
+	if (ishuman(user))
+		var/mob/living/carbon/human/H = user
+		if(H.gloves)
```
Comment (DamianX) [inline code/WorkInProgress/Ported/policetape.dm]: use `get_item_by_slot` instead
call it once and re-use it in the line below this one

File: code/WorkInProgress/Ported/policetape.dm [hunk 7]
```diff
@@ -214,5 +219,148 @@
...
+
+/obj/item/tape/engineering/syndie/proc/spark_and_shock(var/mob/user)
+	if (user && charged)
+		spark(src, 5)
+		return shock(user, 50)
+	else // No user, or not charged
+		return FALSE
+
+/obj/item/tape/engineering/syndie/proc/shock(var/mob/user, var/damage)
+	if (!user || !istype(user, /mob/living))
```
Comment (DamianX) [inline code/WorkInProgress/Ported/policetape.dm]: first check is unnecessary

**Review by DamianX — CHANGES_REQUESTED** [review]

File: code/datums/uplink_item.dm (hunk 5 above)
Comment (ShiftyRail) [reply code/datums/uplink_item.dm]: The lore reason for why they carry a toolbox of tools in their backpack of course

**Comment by DamianX** [comment]
When I testmerged this the cuffs applied by the security variant were invisible, is that still a thing?
Also update the changelog

**Comment by AI-Is-Rogue** [comment]
Would anything happen if you actually cuffed someone with the tape itself?

**Comment by Exxion** [comment]
The description of the police tape still says it explodes

**Comment by ShiftyRail** [comment]
Ah fuck Git why must you hurt me this way

**Comment by ShiftyRail** [comment]
I'll unfuck this PR this evening

File: code/WorkInProgress/Ported/policetape.dm [hunk 8]
```diff
@@ -207,5 +219,149 @@
...
+				H.equip_to_slot(cuffs, slot_handcuffed)
+		return FALSE
+
+
+/obj/item/tape/police/syndie/examine(mob/user)
+	. = ..()
+	if (get_dist(user, src) < 3)
+		to_chat(user, "<span class = 'warning'>This one looks heavier than the usual.</span>")
+
+// -- Syndie atmos tape : shocks and sparks you (useful for lighting those plasma fires)
```
Comment (DamianX) [inline code/WorkInProgress/Ported/policetape.dm]: wrong comment, this is the engineering one

**Comment by DamianX** [comment]
Also the changelog is still wrong ree

**Comment by SonixApache** [comment]
for the record; deleting all your commits is generally not how you should do it

**Comment by Exxion** [comment]
Gotta learn how 2 rebase

**Comment by ShiftyRail** [comment]
@SonixApache @Exxion I just saved work, deleted all commits and then pulled from upstream and committed again
It's a bit unga bunga but it works

**Comment by SonixApache** [comment]
[insert xkcd comic here]

File: code/WorkInProgress/Ported/policetape.dm [hunk 9]
```diff
@@ -207,5 +219,149 @@
...
+	icon_base = "engineering"
+	tape_type = /obj/item/tape/engineering/syndie
+
+/obj/item/taperoll/syndie/preattack(atom/target, mob/user, proximity_flag, click_parameters)
+	if (charges_left & (istype(target, /obj/machinery/door/airlock) || istype(target, /obj/machinery/door/firedoor)))
+		charges_left--
+		if (!(charges_left))
+			to_chat(user, "<span class = 'warning'>There is no tape left.</span>")
+			qdel(src)
+			return
```
Comment (DamianX) [inline code/WorkInProgress/Ported/policetape.dm]: This should return TRUE

Comment (ShiftyRail) [reply code/WorkInProgress/Ported/policetape.dm]: I trust you on this, but why ?

Comment (DamianX) [reply code/WorkInProgress/Ported/policetape.dm]: Otherwise it won't prevent [whatever is being hit]'s attackby from running

**Comment by SonixApache** [comment]
it doesn't fit IN THE BEEEEEEEEEELT

**Pull request merged**